src/Factory.php

```diff
@@ -19,21 +19,44 @@ final class Factory
      *
      * This method should usually only be called once at the beginning of the program.
      *
+     * @deprecated Use Loop::get instead
+     *
      * @return LoopInterface
      */
     public static function create()
+    {
+        $loop = self::construct();
+
+        Loop::set($loop);
```

Comment:
This side effect of a factory is pretty much unexpected IMO.

Reply:
Main reason for adding that was to support mixed use cases where both the Factory and the Loop object are used.

Reply:
> This side effect of a factory is pretty much unexpected IMO.

I agree that this may appear odd on first sight. Like @WyriHaximus pointed out, we're using this as a starting point to be able to gather more feedback. Right now this has no effect on any consumer because the `Loop::set()` is entirely internal only and the `Loop::get()` uses the very same `Factory::create()` method. Expect a cleaner approach not too far in the future once we have a public `Loop::set()` method in a separate PR :shipit: